src/mode/modes.ts

```diff
@@ -73,7 +73,11 @@ export class SearchInProgressMode extends Mode {
     }
     const leadingChar =
       vimState.globalState.searchState.searchDirection === SearchDirection.Forward ? '/' : '?';
-    return `${leadingChar}${vimState.globalState.searchState!.searchString}`;
+
+    let stringWithCursor = vimState.globalState.searchState!.searchString.split('');
+    stringWithCursor.splice(vimState.statusBarCursorCharacterPos, 0, '|');
```

Comment:
As per your comment in the `Get()` method, isn't the pipe removed?

Reply:
yeah it is mainly doing that for the search I should probably move the trimming happening in the `Get()` method to a different method eg `GetTrimmed()` since it is not just getting the status bar text.